sqlair.go

```diff
@@ -92,15 +97,98 @@ type Iterator struct {
 	started bool
 }
 
+// A Least Recently Used Cache using a doubly linked list.
+type lruCache struct {
+	ll   *list.List
+	c    map[index]*list.Element
+	size int
+}
+
+// The cache key.
+type index struct {
```

Comment:
This comment feels so short it could be encoded in the type name. Perhaps this type could be renamed to `cacheKey`?

Reply:
True, that makes it clearer.